jxls-poi/src/main/java/org/jxls/common/PoiExceptionLogger.java

```diff
@@ -1,38 +1,82 @@
-    private static final Logger logger3 = LoggerFactory.getLogger(EachCommand.class);
+public class PoiExceptionLogger implements JxlsLogger {
 
     @Override
-    public void handleCellException(Exception e, String cell, String contextKeys) {
-        logger1.error("Failed to write a cell with {} and context keys {}", cell, contextKeys, e);
+    public void handleCellException(Exception e, String cell, Context context) {
+        error(e, "Failed to write a cell with " + cell);
     }
 
     @Override
     public void handleFormulaException(Exception e, String cell, String formula) {
-        logger1.error("Failed to set formula = " + formula + " into cell = " + cell, e);
+        error(e, "Failed to set formula = " + formula + " into cell = " + cell);
     }
 
     @Override
     public void handleTransformException(Exception e, String sourceCell, String targetCell) {
-        logger2.error("Failed to transform " + sourceCell + " into " + targetCell, e);
+        error(e, "Failed to transform " + sourceCell + " into " + targetCell);
     }
 
     @Override
     public void handleUpdateRowHeightsException(Exception e, int sourceRow, int targetRow) {
-        logger2.error("Failed to update row height for src row={} and target row={}", sourceRow, targetRow, e);
+        error(e, "Failed to update row height for src row=" + sourceRow + " and target row=" + targetRow);
     }
 
     @Override
     public void handleEvaluationException(Exception e, String cell, String expression) {
-        logger3.warn("Failed to evaluate collection expression {}", expression, e);
+        warn(e, "Failed to evaluate collection expression " + expression);
+    }
+
+    @Override
+    public void handleGetObjectPropertyException(Exception e, Object obj, String propertyName) {
+        warn(e, "Failed to get property '" + propertyName + "' of object " + obj);
+    }
+
+    @Override
+    public void handleSetObjectPropertyException(Exception e, Object obj, String propertyName, String propertyValue) {
+        warn(e, "Failed to set property '" + propertyName + "' to value '" + propertyValue + "' for object " + obj);
+    }
+    
+    @Override
+    public void handleSheetNameChange(String invalidSheetName, String newSheetName) {
+        info("Change invalid sheet name " + invalidSheetName + " to " + newSheetName);
+    }
+
+    @Override
+    public void debug(String msg) {
+    }
+
+    @Override
+    public void info(String msg) {
+        System.out.println("JXLS [INFO] " + msg);
+    }
+
+    @Override
+    public void warn(String msg) {
+        write("WARN", msg, null);
+    }
+
+    @Override
+    public void warn(Throwable e, String msg) {
+        write("WARN", msg, e);
+    }
+
+    @Override
+    public void error(String msg) {
+        write("ERROR", msg, null);
+    }
+
+    @Override
+    public void error(Throwable e, String msg) {
+        write("ERROR", msg, e);
+    }
+    
+    protected void write(String level, String msg, Throwable e) {

```

Comment:
The new Jxls logging system is simple but seems to be very limited and I wonder if we are not reinventing the wheel here?
E.g. how one would go if they would like to have all the Jxls logging go to specific file, disable certain logging levels, have automatic file rotation depending on size etc. And if we want to support any of this, should we perhaps consider using the standard Java logging interface from `java.util.logging` instead of a custom solution?

Reply:
I still think it's good to have something between the Jxls code and the logging framework. And that's JxlsLogger. In the past it was a problem to control the logging (vs. throwing exceptions).

A Jxls user can copy e.g. JxlsLoggerForSlf4j.java from our homepage to his workspace and add the slf4j dependencies.

We use PoiExceptionThrower (for standard cases) and someone catch those messages and will log (or display) them.

Reply:
Oh, I get the idea now. It's fine to have a middleman between a logger framework and Jxls. I just want us to ensure Jxls can be integrated into other apps that might be using a different logging system.